apps/etterlatte-beregning/src/main/kotlin/beregning/AvkortingRepository.kt

```diff
@@ -8,28 +12,80 @@ import javax.sql.DataSource
+                Avkorting(
+                    behandlingId = behandlingId,
+                    avkortinggrunnlag,
+                    beregningEtterAvkorting = emptyList(),
+                    tidspunktForAvkorting = Tidspunkt.now() // TODO sette ved beregningEtterAvkorting
                 )
-            ),
-            beregningEtterAvkorting = listOf(),
-            tidspunktForAvkorting = Tidspunkt.now()
-        )
+            }
+        }
     }
 
-    fun lagreAvkortingGrunnlag(behandlingId: UUID, avkortingGrunnlag: AvkortingGrunnlag): Avkorting {
-        return Avkorting(
-            behandlingId = behandlingId,
-            avkortingGrunnlag = listOf(avkortingGrunnlag),
-            beregningEtterAvkorting = listOf(),
-            tidspunktForAvkorting = Tidspunkt.now()
-        )
+    fun lagreEllerOppdaterAvkortingGrunnlag(behandlingId: UUID, avkortingGrunnlag: AvkortingGrunnlag): Avkorting {
+        using(sessionOf(dataSource)) { session ->
+            session.transaction { tx ->
+                queryOf(
+                    statement = """
+                        DELETE FROM avkortinggrunnlag WHERE behandlingId = :behandlingId::UUID
+                    """.trimIndent(),
+                    paramMap = mapOf("behandlingId" to behandlingId)
+                ).let { query ->
+                    tx.run(query.asUpdate)
+                }
+
+                queryOf(
+                    statement = """
+                        INSERT INTO avkortinggrunnlag(
+                        id, behandlingId, datoFOM, datoTOM, aarsInntekt, gjeldendeAar, spesifikasjon
+                        ) VALUES (
+                        :id::UUID, :behandlingId::UUID, :datoFOM::TEXT, :datoTOM::TEXT, 
+                        :aarsInntekt::BIGINT, :gjeldendeAar::INTEGER, :spesifikasjon::TEXT
+                        )
+                    """.trimIndent(),
+                    paramMap = mapOf(
+                        "id" to UUID.randomUUID(),
+                        "behandlingId" to behandlingId,
+                        "datoFOM" to avkortingGrunnlag.periode.fom.toString(),
+                        "datoTOM" to avkortingGrunnlag.periode.tom.toString(),
+                        "aarsInntekt" to avkortingGrunnlag.aarsInntekt,
+                        "gjeldendeAar" to avkortingGrunnlag.gjeldendeAar,
+                        "spesifikasjon" to avkortingGrunnlag.spesifikasjon
+                    )
+                ).let { tx.run(it.asUpdate) }
+            }
+        }
+        return hentAvkortingUtenNullable(behandlingId)
     }
+
+    private fun hentAvkortingUtenNullable(behandlingId: UUID): Avkorting =
+        hentAvkorting(behandlingId) ?: throw Exception("Uthenting av avkorting for behandling $behandlingId feilet")
+
+    private fun Row.toAvkortinggrunnlag() = AvkortingGrunnlag(
+        periode = Periode(
+            fom = YearMonth.parse(string("datoFOM")),
+            tom = string("datoTOM").let {
```

Comment:
Dette vil vel feile siden `string()` forventer at verdien _ikke_ er null? 

Kan uansett forenkles til `stringOrNull("datoTOM")?.let { YearMonth.parse(it) }`

Reply:
Ja var litt rask her men endrer fra TEXT til Date og bruker SqlDateOrNull